helita/sim/bifrost.py

```diff
@@ -329,44 +360,196 @@ def _get_simple_var_xy(self, var, order='F', mode='r'):
         return np.memmap(filename, dtype=self.dtype, order=order, offset=offset,
                          mode=mode, shape=(self.nx, self.ny))
 
-
-    def _get_composite_var(self, var):
+    def boundcut(self, name,var):
```

Comment:
This function should only be called when `params['boundarychk'] = 1`, but right now it is called all the time. Besides, it should not even exist. You are getting these problems because you called `init_cstagger` with `self.nz`, but you should have used `self.nzb`, which includes the additional boundary points.